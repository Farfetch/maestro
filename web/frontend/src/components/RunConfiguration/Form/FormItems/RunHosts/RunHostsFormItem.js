import { MinusCircleOutlined, PlusOutlined } from "@ant-design/icons";
import { Button, Col, Form, Input, Row } from "antd";

const RunHostsFormItem = () => (
  <>
    <Form.List name="hosts">
      {(fields, { add, remove }) => (
        <>
          {fields.map((field) => (
            <Row key={field.key} align="middle" gutter={[6, 0]}>
              <Col flex="auto">
                <Row gutter={[24, 0]}>
                  <Col span={12}>
                    <Form.Item
                      {...field}
                      label="Host"
                      name={[field.name, "host"]}
                      fieldKey={[field.fieldKey, "host"]}
                      rules={[{ required: true, message: "Missing host" }]}
                    >
                      <Input />
                    </Form.Item>
                  </Col>
                  <Col span={12}>
                    <Form.Item
                      {...field}
                      label="IP"
                      name={[field.name, "ip"]}
                      fieldKey={[field.fieldKey, "ip"]}
                      rules={[{ required: true, message: "Missing IP" }]}
                    >
                      <Input />
                    </Form.Item>
                  </Col>
                </Row>
              </Col>

              <Col flex="20px" style={{ margin: "30px 0 24px 0" }}>
                <MinusCircleOutlined onClick={() => remove(field.name)} />
              </Col>
            </Row>
          ))}

          <Button
            type="dashed"
            onClick={() => add()}
            block
            icon={<PlusOutlined />}
          >
            Add custom host
          </Button>
        </>
      )}
    </Form.List>
  </>
);

export default RunHostsFormItem;
