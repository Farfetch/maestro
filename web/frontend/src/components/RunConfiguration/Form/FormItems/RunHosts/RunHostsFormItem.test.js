import { cleanup, render, waitFor } from "@testing-library/react";
import { Form } from "antd";
import React from "react";

import RunHostsFormItem from "./RunHostsFormItem";

afterEach(cleanup);

describe("components/RunConfiguration/Form/RunHostsFormItem", () => {
  test(`should render RunHostsFormItem component`, async () => {
    const { container } = render(
      <Form>
        <RunHostsFormItem />
      </Form>
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });
});
