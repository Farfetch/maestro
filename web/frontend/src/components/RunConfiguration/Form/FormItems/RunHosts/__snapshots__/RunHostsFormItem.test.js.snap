// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunConfiguration/Form/RunHostsFormItem should render RunHostsFormItem component 1`] = `"<div><form class=\\"ant-form ant-form-horizontal css-dev-only-do-not-override-nnuwmp\\"><button type=\\"button\\" class=\\"ant-btn css-dev-only-do-not-override-nnuwmp ant-btn-dashed ant-btn-block\\"><span class=\\"ant-btn-icon\\"><span role=\\"img\\" aria-label=\\"plus\\" class=\\"anticon anticon-plus\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"plus\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z\\"></path><path d=\\"M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z\\"></path></svg></span></span><span>Add custom host</span></button></form></div>"`;
